File: Blind75/424. Longest Repeating Character Replacement/424. Longest Repeating Character Replacement.cs
public class Solution {
    public int CharacterReplacement(string s, int k) {
        int n = s.Length;
        Dictionary<char,int> freq = new Dictionary<char,int>();
        int mostFreq = 0;
        int left = 0;
        int right = 0;
        int res = 0;
        while(right<n){
            if(freq.ContainsKey(s[right])){
                freq[s[right]]++;
            }else freq.Add(s[right],1);
            //maintain the global highest freq
            mostFreq = Math.Max(mostFreq, freq[s[right]]);
            while( right-left+1 -mostFreq > k){
                freq[s[left]]--;
                left++;
            }
            res = Math.Max(res, right-left+1);
            right++;
        }
        return res;
    }
}

File: Blind75/424. Longest Repeating Character Replacement/424. Longest Repeating Character Replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        unordered_map<char,int> freq;
        int res = 0;
        int n = s.size();
        int left = 0;
        int right = 0;
        int max_freq = 0;
        while(right<n){
            freq[s[right]]++;
            max_freq = max(max_freq, freq[s[right]]);
            //cout<<left<<", "<<right<<", k = "<<(right-left+1)-freq[s[left]]<<endl;
            if( (right-left+1)-max_freq > k ){ // more than k needs to be change
                freq[s[left]]--;
                left++;
            }
            res = max(res, right-left+1);
            right++;
        }
        return res; 
    }
};
